lib/jvm.sh

```diff
@@ -28,7 +28,8 @@ else
 fi
 
 get_jdk_version() {
-  local appDir="${1:?}"
+  local appDir
+  appDir="${1:?}"
```

Comment:
so, I could have sworn I learned from shellcheck not to declare and initialize shell variables at the same time, but couldn't find the reference nor get it to fail.  On the other hand, the file is inconsistent about this.

Reply:
Shellcheck does warn about that, but it only applies to cases where a command is run (rather than a plain variable assignment):
https://www.shellcheck.net/wiki/SC2155

Reply:
I removed all the changes except for the unbound variables since that's the one I was really interested in

and I added a commit to run test tasks with `set -eou pipefail` in case you were interesting in testing in a stricter environment than you run in. I'm happy to remove that.

Reply:
Hey @bf4, I just ran CI again and noticed that `set -eou pipefail` causes our unit tests to fail since they don't run in `bash`. Otherwise, everything is passing on CI now 🎉 . It's probably the easiest to remove the `set -eou pipefail` so we can ship this soon :)

Reply:
Thanks @Malax I reverting the last commit (since I figure that's better PR etiquette vs remove and force push). Let me know  how I can help you help me help us :)

Reply:
Thanks! ✋🏻